exp/api/v1beta1/azuremanagedmachinepool_types.go

```diff
@@ -65,6 +65,10 @@ type AzureManagedMachinePoolSpec struct {
 	// Scaling specifies the autoscaling parameters for the node pool.
 	// +optional
 	Scaling *ManagedMachinePoolScaling `json:"scaling,omitempty"`
+
+	// MaxPods specifies the kubelet --max-pods configuration for the node pool.
+	// +optional
+	MaxPods *int32 `json:"maxPods,omitempty"`
```

Comment:
you'll also need to add this new type to conversion from v1alpha3 and v1alpha4 to preserve the field during round trip conversions, see an example here: https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1815/files#diff-45673dfe0beba334668369fcf88dcfb9a531245efe19cfb6d413e9bb6420e04dR41